core/src/main/scala/org/apache/spark/resource/ResourceInformation.scala

```diff
@@ -39,6 +39,18 @@ class ResourceInformation(
     val name: String,
     val addresses: Array[String]) extends Serializable {
 
+  def + (other: ResourceInformation): ResourceInformation = {
```

Comment:
Note this may generates much more intermediate `ResourceInformation` objects. I do this because `ResourceInformation` yet is immutable. It may be better if we could make it be mutable in the future.

Reply:
like I mention above we don't want to make this mutable since end user facing and I think you should use different class to track it there to be more efficient